File: supportlib/src/main/java/br/com/helpdev/supportlib/xml/ControllerSimpleXML.java
package br.com.grupocriar.swapandroid.xml;

import org.simpleframework.xml.Serializer;
import org.simpleframework.xml.core.Persister;

import java.io.StringWriter;

/**
 * Created by Guilherme Biff Zarelli on 11/29/17.
 */

public class ControllerSimpleXML {
    private ControllerSimpleXML() {
        throw new RuntimeException("No ControllerSimpleXML");
    }

    @SuppressWarnings("unchecked")
    public static <T> T xmlToObject(Class tClass, String xml) throws Exception {
        Serializer serializer = new Persister();
        return (T) serializer.read(tClass, xml);
    }

    public static String objectToXml(Object object) throws Exception {
        Serializer serializer = new Persister();
        StringWriter sw = new StringWriter();
        serializer.write(object, sw);
        return sw.toString();
    }
}

File: supportlib/src/main/java/br/com/helpdev/supportlib/xml/ControllerSimpleXML.kt
package br.com.helpdev.supportlib.xml

import org.simpleframework.xml.Serializer
import org.simpleframework.xml.core.Persister

import java.io.StringWriter

/**
 * Created by Guilherme Biff Zarelli on 11/29/17.
 */

class ControllerSimpleXML {
    companion object {

        @Throws(Exception::class)
        fun <T> xmlToObject(tClass: Class<*>, xml: String): T {
            val serializer = Persister()
            return serializer.read(tClass, xml) as T
        }

        @Throws(Exception::class)
        fun objectToXml(`object`: Any): String {
            val serializer = Persister()
            val sw = StringWriter()
            serializer.write(`object`, sw)
            return sw.toString()
        }
    }
}
